internal/operations/operation_updater.go

```diff
@@ -310,21 +310,33 @@ func (ou *operationUpdater) doUpdate(ctx context.Context, update *core.Operation
 	}
 
 	// Match a TX we already retrieved, if found add a specified Blockchain Transaction ID to it
+	var txnIDStr string
+	var idempotencyKeyStr string
 	var tx *core.Transaction
-	if op.Transaction != nil && update.BlockchainTXID != "" {
+	if op.Transaction != nil {
 		for _, candidate := range transactions {
 			if op.Transaction.Equals(candidate.ID) {
 				tx = candidate
+				txnIDStr = candidate.ID.String()
+				idempotencyKeyStr = string(candidate.IdempotencyKey)
```

Comment:
BTW incase anyone else looks at this, IdempotencyKey is not a pointer just a string type  so safe to cast to string